lib/archethic/bootstrap.ex

```diff
@@ -300,19 +301,19 @@ defmodule Archethic.Bootstrap do
       |> Enum.reject(&(&1.first_public_key == Crypto.first_node_public_key()))
 
     # In case node had lose it's DB, we ask the network if the node chain already exists
-    case Crypto.next_node_public_key()
-         |> Crypto.derive_address()
-         |> TransactionChain.fetch_size_remotely(closest_nodes) do
-      {:ok, length} when length > 0 ->
-        Crypto.set_node_key_index(length)
+    {:ok, length} =
+      Crypto.first_node_public_key()
+      |> Crypto.derive_address()
+      |> TransactionChain.fetch_size_remotely(closest_nodes)
 
-      _ ->
-        :skip
-    end
+    Crypto.set_node_key_index(length)
 
     tx =
       TransactionHandler.create_node_transaction(ip, port, http_port, transport, reward_address)
 
+    # Reset node index as it will be handled by node transaction replication during self repair
```

Comment:
To avoid that, we could use `Transaction.new/3` and pass as argument in `create_node_transaction` the index